client/src/pages/boxes/ListBoxesPage/ListBoxesPage.tsx

```diff
@@ -118,14 +118,14 @@ export const ListBoxesPage = () => {
 
   const mutation = useMutation({
     mutationFn: () =>
-      fetcher<Blob>(`${APIManager.baseAPIRUrl}/charityBoxes/csv`, { returnBlob: true }),
+      fetcher<Blob>(`${APIManager.baseAPIRUrl}/charityBoxes/xlsx`, { returnBlob: true }),
     onSuccess: (data) => {
       const url = window.URL.createObjectURL(new Blob([data]));
       const link = document.createElement('a');
       link.href = url;
       link.setAttribute(
         'download',
-        `charity_boxes${(Math.random() * 10000000000).toFixed(0)}.csv`,
+        `charity_boxes${(Math.random() * 10000000000).toFixed(0)}.xlsx`,
```

Comment:
WTFs per minute skyrocketed. I propose we add a time+date to the filename (12_30_22_26_11_2025) and shorten the random number